File: 2022/ShortDeal/app/src/main/java/com/creator/shortdeal/adapters/PageAdapter.java
package com.creator.shortdeal.adapters;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Matrix;
import android.support.annotation.NonNull;
import android.support.v4.view.PagerAdapter;
import android.support.v4.view.ViewPager;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.creator.shortdeal.models.Image;
import com.creator.shortdeal.util.BitmapUtil;

import java.util.ArrayList;

public class PageAdapter extends PagerAdapter {
	Context context;
	private ArrayList<Image> mSelectedImgages;

	public PageAdapter(Context context, ArrayList<Image> selectedImages){
		this.context=context;

		this.mSelectedImgages = selectedImages;
	}

	@Override
	public int getCount() {
		return mSelectedImgages.size();
	}

	@Override
	public boolean isViewFromObject(View view, Object object) {
		return view == ((ImageView) object);
	}

	@Override
	public Object instantiateItem(ViewGroup container, int position) {
		ImageView imageView = new ImageView(context);
//		imageView.setScaleType(ImageView.ScaleType.CENTER_INSIDE);
		imageView.setScaleType(ImageView.ScaleType.FIT_CENTER);
		imageView.setImageBitmap(getImageBitmap(mSelectedImgages.get(position).path));

		Log.d("SeongKwon", (mSelectedImgages.get(position).sequence - 1) + "//" + getCount() + "//");
		((ViewPager) container).addView(imageView, 0);
		return imageView;
	}

	@Override
	public void destroyItem(ViewGroup container, int position, Object object) {
		((ViewPager) container).removeView((ImageView) object);
	}

	public Bitmap getImageBitmap(String path) {
		Bitmap resized = null;
		try {
			// 회전
			Matrix matrix = new Matrix();
			matrix.postRotate(BitmapUtil.GetExifOrientation(path));


            BitmapFactory.Options options = new BitmapFactory.Options();
            options.inSampleSize = 2;
            Bitmap src = BitmapFactory.decodeFile(path, options);

            int width = src.getWidth();
            int height = src.getHeight();
            resized = Bitmap.createBitmap(src, 0, 0, width, height, matrix, true);
		} catch (IllegalArgumentException e) {
			e.printStackTrace();
		}
		return resized;
	}

	public void addItem(Image image, int index) {
		mSelectedImgages.add(index, image);
		notifyDataSetChanged();
	}

	public void removeItem(int index) {
		mSelectedImgages.remove(index);
		notifyDataSetChanged();
	}

	@Override
	public int getItemPosition(@NonNull Object object) {
//		return super.getItemPosition(object);
		return POSITION_NONE;
	}
}

File: 2025/ChanggoLive/app/src/main/java/com/creator/changgolive/adapters/PageAdapter.kt
package com.creator.changgolive.adapters

import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.graphics.Matrix
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import com.creator.changgolive.models.Image
import com.creator.changgolive.util.BitmapUtil

class PageAdapter(var context: Context, private val mSelectedImgages: ArrayList<Image>?) :
    PagerAdapter() {
    override fun getCount(): Int {
        return mSelectedImgages!!.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object` as ImageView
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val imageView = ImageView(context)
        //		imageView.setScaleType(ImageView.ScaleType.CENTER_INSIDE);
        imageView.scaleType = ImageView.ScaleType.FIT_CENTER
        imageView.setImageBitmap(getImageBitmap(mSelectedImgages!![position].path))
        Log.d(
            "SeongKwon",
            (mSelectedImgages[position].sequence - 1).toString() + "//" + count + "//"
        )
        (container as ViewPager).addView(imageView, 0)
        return imageView
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        (container as ViewPager).removeView(`object` as ImageView)
    }

    fun getImageBitmap(path: String?): Bitmap? {
        var resized: Bitmap? = null
        try {
            // 회전
            val matrix = Matrix()
            matrix.postRotate(BitmapUtil.Companion.GetExifOrientation(path).toFloat())
            val options = BitmapFactory.Options()
            options.inSampleSize = 2
            val src = BitmapFactory.decodeFile(path, options)
            val width = src.width
            val height = src.height
            resized = Bitmap.createBitmap(src, 0, 0, width, height, matrix, true)
        } catch (e: IllegalArgumentException) {
            e.printStackTrace()
        }
        return resized
    }

    fun addItem(image: Image, index: Int) {
        mSelectedImgages!!.add(index, image)
        notifyDataSetChanged()
    }

    fun removeItem(index: Int) {
        mSelectedImgages!!.removeAt(index)
        notifyDataSetChanged()
    }

    override fun getItemPosition(`object`: Any): Int {
//		return super.getItemPosition(object);
        return POSITION_NONE
    }
}